New section for Authorization Mechanisms based on WRPRC and EDP (#40)
APTITUDE-Consortium/wp2-trust-specifications: topics/entitlement-policy.md

| `service` | array of MultiLangString | **WRPs** (REQUIRED) | User transparency about service provided, support informed consent | [ETSI TS 119 475 Table 7], [CIR 2025/848 Annex I.8]  |
| `public_body` | boolean | **WRPs** (REQUIRED by CIR 2025/848 Annex I.11) | Public sector body identification for trust assessment | [ETSI TS 119 475 Table 10], [CIR 2025/848 Annex I.11] |
| `dpa` | LegalEntity | **WRPs** (REQUIRED) | Data protection supervisory authority for user rights and accountability | [ETSI TS 119 475 Table 7], [CIR 2025/848 Annex IV.3(g)]  |
| `registry_uri` | string (URL) | **WRPs** (REQUIRED) | Register URL for runtime verification and user opt-in queries | [ETSI TS 119 475 Table 7], [CIR 2025/848 Article 3(5)], [ARF RPRC_18]  |
| `support_uri` | string[^1] | **WRPs** (REQUIRED) | Support contact points for user rights including data deletion requests | [ETSI TS 119 475 Table 10], [CIR 2025/848 Annex I.7(a)] |
| `act` | array of WalletRelyingParty | **WRPs** (OPTIONAL) | Intermediary entities disclosure for data flow transparency and trust assessment | [ETSI TS 119 475 Table 10], [CIR 2025/848 Annex I.14] |

> | `credentials[].claim[]` | array of Claim | **RPs** (OPTIONAL) | Attribute overasking prevention - verify requested attributes match registration. If not available, all attributes are requested | [ETSI TS 119 475 Table 9], [ARF RPRC_09], [ARF RPRC_21] |

Reviewer: can we further elaborate on what are this values?

   - Required value: `https://uri.etsi.org/19475/Entitlement/Service_Provider` (per ETSI TS 119 475 Annex A.2, CIR 2025/848)
   - If entitlement check fails:
     - Record verification result is WRONG_ENTITLEMENT
     - Proceed to Step 5 (display results)

2. The WI MUST detect intermediary scenario:
   - Extract RP identifier from WRPAC (subject identifier)
   - Extract RP identifier from presentation request RP info extension (per ARF RPRC_19a: field `id`)
   - Compare identifiers:
     - If identifiers **match** then the direct RP scenario applies, proceed to substep 3
     - If identifiers **do not match** then the intermediary scenario applies (intermediary holds WRPAC, RP info extension refers to intermediated RP), proceed to substep 4

3. **Direct RP scenario**:
   - The WI MUST verify that subject identifier from registration data matches RP identifier from WRPAC.
   - If binding verification fails:
     - Record verification result is BINDING_FAILED
     - Proceed to Step 5 (display results)
   - If binding verification succeeds, proceed to Step 4 (compare attributes).

4. **Intermediary scenario**:

>    - The WI MUST verify that intermediary association from registration data (e.g., `act` array) contains intermediary identifier (from WRPAC).

Reviewer: > [...] (e.g., `act` array)

why e.g.? I think this must be present in the RG if the RP uses an intermediary

Furthermore in TS 119 475 it is a json object with parameters `sub.id` and `sub.name`.

   - If association verification fails:
     - Record verification result is INTERMEDIARY_NOT_AUTHORIZED
     - Proceed to Step 5 (display results)
   - The WI MUST verify that subject identifier from registration data matches intermediated RP identifier from RP info extension.
   - If binding verification fails:
     - Record verification result is BINDING_FAILED
     - Proceed to Step 5 (display results)
   - The WI MUST prepare user display per ARF RPI_07: "[intermediary name] acting on behalf of [intermediated RP name] for [intended use description]".
   - If all verifications succeed, proceed to Step 4 (compare attributes).

**Step 4: Compare Requested vs Registered Attributes**

This step operates on registration data obtained from Step 2a (WRPRC) or Step 2b (Register).

1. The WI MUST extract requested attributes from presentation request:
   - **Remote flow (DCQL)**: Extract from `credential_queries[].claims[]` paths.
   - **Proximity flow**: Extract from device request `namespaces` structure.

2. For each credential type requested:
   1. The WI MUST identify corresponding credential type in registered attributes (from registration data).

>    2. For each attribute requested in that credential type, the WI MUST check if attribute is present in the registered attributes list (per ARF RPRC_21). The matching MUST be **case-sensitive** and **exact**.

Reviewer: Can we specify that the check are:
- (for remote flow) the `credential_queries[].claims[]` and `vct` parameters against the `credentials.claim` array and `credential.meta.vct_values` within the WRPRC;
- (for the proximity flow) the `docRequest.itemRequest.docType` and `docRequest.itemRequest.nameSpaces` parameters of the mdoc request with the `credential.meta.doctype_value` parameter and `credential.claims.` array of the WRPRC.

   - The WI MUST allow user to approve or deny presentation (user override allowed - advisory mechanism)

3. If verification result is FAILED, CERTIFICATE_INVALID, WRONG_ENTITLEMENT, BINDING_FAILED, or INTERMEDIARY_NOT_AUTHORIZED:
   - The WI MUST display per ARF RPRC_17/18: "Could not verify the information registered about this Relying Party"
   - The WI MUST allow user to approve or deny presentation (user override allowed - advisory mechanism)

4. If verification was skipped (user not opted-in):
   - The WI MAY display: "Registration verification was not performed"

After user is informed, proceed to Step 6 (embedded disclosure policy evaluation).

**Step 6: Proceed to Embedded Disclosure Policy Evaluation**

After registration verification (success, failure, or skipped), the WI MUST always proceed to embedded disclosure policy evaluation (see [Embedded Disclosure Policy Enforcement Mechanism](#embedded-disclosure-policy-enforcement-mechanism)). Policy evaluation is executed regardless of registration verification result.

### Embedded Disclosure Policy Enforcement Mechanism

> This mechanism allows Attestation Providers to define policies that restrict which Relying Parties can access specific Attestations. Unlike RP Overasking Prevention (which is opt-in and advisory), policy enforcement is **mandatory** and **cannot be overridden** by the user (per ARF EDP_07, RPA_11). If policy evaluation fails, the WI behaves as if the Attestation does not exist.

Author: You are right, I will fix asap

#### Common Registration Phases

The following steps are in common to all WRPs for which registration is required.

1. **Identity Verification**: Registrar verify identity of WRP according requirements in ETSI TS 119 461. The specific identity proofing level may vary based on entity type and applicable regulatory framework (e.g. QTSP requirements or MS national legislation) and it is out of scope of the piloting.

2. **Registration Record Creation**: Registrar create registration record in national register. Record contain at least:
   - WRP identification information.
   - WRP type (RP, PID Provider, QEAA Provider, PuB-EAA Provider, EAA Provider).
   - Entity-specific capabilities (see [Entity-Specific Specialization](#entity-specific-specialization))

3. **WRPRC Issuance**: If the Member State mandates WRPRC issuance according to CIR 2025/848 Article 8, the Provider of WRPRCs must issue a signed WRPRC containing registered capabilities. If it is not mandated, Wallet Instance may retrieve information from Registrar's online service. 

> #### Entity-Specific Capabilities

Reviewer 2: ```suggestion
Wallet-Relying Parties (WRPs) must register with Registrar in their Member State before they can operate in the EUDI Wallet ecosystem. The Registration Process follows common phases with entity-specific specializations. See Section [Trust Architecture](https://github.com/APTITUDE-Consortium/wp2-trust-specifications/blob/main/topics/trust-architecture.md) for more details.
```

| `service` | array of MultiLangString | **WRPs** (REQUIRED) | User transparency about service provided, support informed consent | [ETSI TS 119 475 Table 7], [CIR 2025/848 Annex I.8]  |
| `public_body` | boolean | **WRPs** (REQUIRED by CIR 2025/848 Annex I.11) | Public sector body identification for trust assessment | [ETSI TS 119 475 Table 10], [CIR 2025/848 Annex I.11] |
| `dpa` | LegalEntity | **WRPs** (REQUIRED) | Data protection supervisory authority for user rights and accountability | [ETSI TS 119 475 Table 7], [CIR 2025/848 Annex IV.3(g)]  |
| `registry_uri` | string (URL) | **WRPs** (REQUIRED) | Registrar API URL for runtime verification and user opt-in queries | [ETSI TS 119 475 Table 7], [CIR 2025/848 Article 3(5)], [ARF RPRC_18]  |
| `support_uri` | string[^1] | **WRPs** (REQUIRED) | Support contact points for user rights including data deletion requests | [ETSI TS 119 475 Table 10], [CIR 2025/848 Annex I.7(a)] |
| `act` | array of WalletRelyingParty | **WRPs** (OPTIONAL) | Intermediary entities disclosure for data flow transparency and trust assessment | [ETSI TS 119 475 Table 10], [CIR 2025/848 Annex I.14] |

> | `credentials[].claim[]` | array of Claim | **RPs** (REQUIRED) | Attribute overasking prevention - verify requested attributes match registration | [ETSI TS 119 475 Table 9], [ARF RPRC_09], [ARF RPRC_21] |

Reviewer 2: ```suggestion
| `credentials[].claim[]` | array of Claim | **RPs** (OPTIONAL) | Attribute overasking prevention - verify requested attributes match registration. If not available, all attributes are requested | [ETSI TS 119 475 Table 9], [ARF RPRC_09], [ARF RPRC_21] |
```

| `status`	| object {idx: string, uri: string} |	**WRPs** (REQUIRED)	| Revocation verification - check WRPRC validity status before using registration data|	[ETSI TS 119 475 GEN-6.2.6.1-04], [ARF RPRC_17]

[^1]: ETSI shows `support_uri` as singular (URL or email), but Looking at Table 1 `supportURI` is related to Subject Alternative Name can have multiple entries. To check if it makes sense to define `support_uri` as array of strings.

> [!NOTE]
>
> See [Section X.Y - WRPRC Detailed Specification] for additional details, including formats, encodings, complete examples, and ETSI mappings.

> [!WARNING]
> The cross reference above should be replaced when the section about WRPRC will be finalized

### Distribution Methods

Relying Party Instance include WRPRC in presentation request for **both proximity and remote presentation flows** (RPRC_19), using extension defined in ETSI TS 119 472-2.
During  the **Proximity flow** (ISO/IEC 18013-5), the WRPRC is included in device request. 

> In the **Remote flow** (OpenID4VP), the WRPRC is included in authorization request. 

Reviewer 2: ```suggestion
Relying Party Instances include the WRPRC in a presentation request for **both proximity and remote presentation flows** (RPRC_19), using extension defined in ETSI TS 119 472-2.
During  the **Proximity flow** (ISO/IEC 18013-5), the WRPRC is included in the device request. 
In the **Remote flow** (OpenID4VP), the WRPRC is included in the authorization request. 
```

The main prurpose is to allow Attestation Provider to control which Relying Parties can access specific Attestation.

Attestation Provider can optionally express Embedded Disclosure policy for Attestations (EDP_01), and, accoridng to the Article 10 of CIR 2024/2979, Wallet Providers ensure that electronic Attestations with common Embedded Disclosure policies (listed in Annex III) can be processed by their Wallet Instances.

If no policy defined, Attestations can be presented to any Relying Party, although they are always subject to user consent.

The Embedded Disclosure Policies are applicable to:
- **QEAAs**. 
- **PuB-EAAs**. 
- **Non-qualified EAAs**.

They are **not applicable to PIDs**: EUDIW Regulation doesn't provide any requirement for PIDs to contain Embedded Disclosure Policy (EDP_01 note).

The main **use cases** that can be enabled by Embedded Disclosure Policies are:
- Restrict access to sensitive Attestations to authorized RPs only.
- If a sector-specific CAs exists, the AP may implement sector-specific access control (e.g., only public sector RPs)

> - MS-specific access control (only RPs registered within a specific MEmber State)

Reviewer 2: ```suggestion
- If a sector-specific CAs exists, the AP may implement sector-specific access control (e.g., only public sector RPs).
- MS-specific access control (only RPs registered within a specific Member State).
```

In particular, the use cases above are enabled using the policy types described in the following section. 

### Policy Types

Annex III of CIR 2024/2979 define three common embedded disclosure policy types that Wallet Instance must support.

1. **No Policy**: Attestation has no embedded disclosure policy. Wallet Instance can present Attestation to any Relying Party, subject only to user approval.
2. **Authorized Relying Parties Only Policy**: Attestation can be presented only to Relying Parties in authorized list defined by Attestation Provider. According to (EDP_02), this list contains the **EU-wide unique identifiers** of authorized Relying Parties. The format is the same as the unique identifier in WRPRC.
3. **Specific Root of Trust Policy**: Attestation can be presented only to Relying Parties whose WRPACs signed by specific trusted Certificate Authorities. According to (EDP_03), the list contains the **root or intermediate certificates** used for signing Relying Party WRPACs, and must be in X.509 format.

### Policy Structure and Encoding

> [!WARNING]

> > The Embedded Disclosure Policy structure and parameter names defined in this section are not based on normative references neither on technical standards. ARF requirement EDP_08 state that Commission shall create technical specification for embedded disclosure policy format. However this specification is **not yet published**. The content of this section is based on EDP_02, EDP_03, EDP_05, but may differ from final technical specification.

Reviewer 2: ```suggestion
> The Embedded Disclosure Policy structure and parameter names defined in this section are not based on normative references neither on technical standards. ARF requirement EDP_08 state that Commission shall create technical specification for Embedded Disclosure Policy format. However this specification is **not yet published**. The content of this section is based on EDP_02, EDP_03, EDP_05, but may differ from final technical specification.
```

| `type` | string (enum) | REQUIRED. Certificate type in PKI chain. Values: `"root"` (root CA certificate) or `"intermediate"` (intermediate CA certificate between root and leaf). Leaf certificates (RP WRPACs) are not included in policy | [ARF EDP_03], [EDP_03 HLR] |

In the following sections non-normative examples are provided.

#### Example 1: Authorized Relying Parties Only

```json
{
  "policy_type": "authorized_relying_parties_only",
  "version": "1.0",
  "authorized_parties": [
    "VATIN:IT-12345678901",
    "EORI:FR123456789012345"
  ],
  "policy_url": "https://provider.example.com/policies/medical-data"
}
```

#### Example 2: Specific Root of Trust

```json
{
  "policy_type": "specific_root_of_trust",
  "version": "1.0",
  "trusted_certificates": [
    {
      "certificate": "MIICertDataBase64==",
      "type": "root"
    },
    {
      "certificate": "MIIDifferentCertBase64==",
      "type": "intermediate"
    }
  ],
  "policy_url": "https://provider.example.com/policies/trusted-cas"
}
```

### Policy Distribution via Credential Issuer Metadata

> According to ARF Discussion Topic D, the **primary distribution method** for Embedded disclosure policies should be including it in **Credential Issuer metadata**.

Reviewer 2: ```suggestion
According to ARF Discussion Topic D, the **primary distribution method** for Embedded Disclosure Policies should be including it in **Credential Issuer metadata**.
```
